ArduCopter/toy_mode.cpp

```diff
@@ -1080,4 +1096,70 @@ void ToyMode::arm_check_compass(void)
     }
 }
 
+/*
+  handle take of cmd
+ */
+void ToyMode::handle_takeoff_cmd(bool takeoff_btn,bool takeoff_cancel)
+{
+    const uint32_t takeoff_delay_ms = 3000;
+    
+    //initiate takeoff cmd
+    if (takeoff_state == TAKEOFF_ON_GROUND && !copter.motors->armed() && takeoff_btn) {
+        takeoff_state = TAKEOFF_INITIATE;
+        takeoff_last_press_ms = AP_HAL::millis();
+    }
+    
+    switch (takeoff_state) {
+    case TAKEOFF_INITIATE:
+        
+        if (!copter.motors->armed()) {
+            action_arm();
+        }
+        
+        if (!copter.init_arm_motors(true) && (copter.control_mode == LOITER)) {
```

Comment:
I saw this example and thought it would do the same behaviour as if the user push the throttle up  - https://github.com/SkyRocketToys/ardupilot/blob/czar-launch-cmd/ArduCopter/toy_mode.cpp#L420. 

but did not go through and fully understand the difference between copter.init_arm_motors(true), arm_check_compass(), and  action_arm(). 

however, from doing some test between these functions - if I only rely on action_arm() I find that it will not arm if other things are not initialised  but when I used "copter.init_arm_motors(true)" it forcefully arms it and only then will take off. So when the user press the takeoff button it would arm and take off but i find that it will drop after the takeoff script finishes. This is from testing from the physical copter. In SITL however it would consistently takeoff at 2m.

Maybe I should put a check where it checks for the initialisation to complete before going to TAKEOFF_ARMED state. At least the user would see the props spin when the user presses the takeoff button.